File: frontend/src/types/ci/job.ts
// Job
export interface Job {
	id: string
	pipeline_run_id: string
	name: string
	status: 'waiting' | 'running' | 'success' | 'failed' | 'faulted' | 'skipped' | 'canceled'
	started_at?: string
	finished_at?: string
	error_message?: string
}

// JobLog
export interface JobLog {
	id: string
	job_id: string
	content: string
	created_at: string
}

// Artifact
export interface Artifact {
	id: string
	pipeline_run_id: string
	job_name: string
	type: 'docker_image' | 'file'
	name: string
	path?: string
	created_at: string
}

export const jobStatusColors: Record<string, string> = {
	waiting: 'default',
	running: 'processing',
	success: 'success',
	failed: 'error',
	faulted: 'error',
	skipped: 'warning',
	canceled: 'warning',
};

File: backend/src/pomelo_orbit/interfaces/api/ci/dto/job.py
"""作业相关 DTO"""

from datetime import datetime

from pydantic import BaseModel


class JobResp(BaseModel):
    id: str
    pipeline_run_id: str
    name: str
    status: str
    parent_job_id: str | None = None
    started_at: datetime | None = None
    finished_at: datetime | None = None
    error_message: str | None = None

    model_config = {"from_attributes": True}


class JobLogResp(BaseModel):
    id: str
    job_id: str
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}
